go/src/meguca/common/posts.go

```diff
@@ -47,21 +47,30 @@ type Thread struct {
 // Post is a generic post exposed publically through the JSON API. Either OP or
 // reply.
 type Post struct {
-	Editing  bool      `json:"editing,omitempty"`
-	Banned   bool      `json:"banned,omitempty"`
-	Deleted  bool      `json:"deleted,omitempty"`
-	Sage     bool      `json:"sage,omitempty"`
-	ID       uint64    `json:"id"`
-	Time     int64     `json:"time"`
-	Body     string    `json:"body"`
-	Flag     string    `json:"flag,omitempty"`
-	PosterID string    `json:"posterID,omitempty"`
-	Name     string    `json:"name,omitempty"`
-	Trip     string    `json:"trip,omitempty"`
-	Auth     string    `json:"auth,omitempty"`
-	Links    []Link    `json:"links,omitempty"`
-	Commands []Command `json:"commands,omitempty"`
-	Image    *Image    `json:"image,omitempty"`
+	Editing     bool      `json:"editing,omitempty"`
+	Banned      bool      `json:"banned,omitempty"`
+	Deleted     bool      `json:"deleted,omitempty"`
+	MeidoVision bool      `json:"meido_vision,omitempty"`
+	Sage        bool      `json:"sage,omitempty"`
+	ID          uint64    `json:"id"`
+	Time        int64     `json:"time"`
+	Body        string    `json:"body"`
+	Flag        string    `json:"flag,omitempty"`
+	PosterID    string    `json:"posterID,omitempty"`
+	Name        string    `json:"name,omitempty"`
+	Trip        string    `json:"trip,omitempty"`
+	Auth        string    `json:"auth,omitempty"`
+	Links       []Link    `json:"links,omitempty"`
+	Commands    []Command `json:"commands,omitempty"`
+	Image       *Image    `json:"image,omitempty"`
+	ModLog      [3]*ModLogEntry
+}
+
+// ModLogEntry is a smaller version of auth.ModLogEntry for use in Post
```

Comment:
Then why not embed this smaller struct into auth.ModLogEntry? Also, add json struct tags.

Reply:
Because of circular imports.
The struct tags aren't (AFAIK) needed, and may actually be detrimental since this array is only populated on getPost

Reply:
I remember why I couldn't just use ModLogEntry, and why I had to do this now.
It was because of types, and circular imports.
Basically, what I had going before was an array of 3 (auth) ModLogEntry(s) embedded in Post, but when I wanted to use it in I think templates, I got circular imports.
I made a smaller version with only what I needed in Posts instead to avoid that.
I wish I could have casted, but unfortunately we can't cast a type that isn't imported.
I'll look into it again, but I guess I want your thoughts on this.

Reply:
Also because I would have to import "meguca/auth" into common, and that causes circular imports.
I think that was the main reason, actually.

Reply:
I may be able to fix this by forwarding from auth in common/posts like common/websockets

Reply:
No, unfortunately no matter what I do, it just doesn't work.
I can technically get rid of it by getting rid of the type and using an empty interface, but that opens up a whole other can of worms.
Sorry, not sure I can change this one.

Reply:
Okay, but I'll have to move `ModerationAction` to common too.

Reply:
Also, why are you using `[3]*ModLogEntry` ? Just use `[]ModLogEntry`.

Reply:
You told me to use a pointer to each for one.
Also I'd rather keep it to 3 `ModLogEntry`, it's not my intent to show every single entry, just the first of each type, if applicable.

Reply:
This will marshal to JSON as empty entries though. I meant using a pointer, if you only show one mod log entry. Using a slice is best for a collection of variable size. You can still only read up to 3 entries into it. This also adds needless size to the Post struct.